app/assets/javascripts/application.js.erb

```diff
@@ -464,13 +410,81 @@ class _LobstersFunction {
         li.querySelector(".flagger").textContent = "flag";
       } else if (action == "flag") {
         li.querySelector(".flagger").textContent = "unflag";
-      if (thingType == "comment") {
-        li.querySelector(".reason").innerHTML = "| " + Lobster.commentFlagReasons[reason].toLowerCase();
+    }
+
+    fetchWithCSRF("/stories/" + li.getAttribute("data-shortid") + "/" + action, {
+      method: 'post',
+      body: formData });
+  }
+
+  voteComment(voterEl, point, reason) {
+    if (!Lobster.curUser)
+      return Lobster.bounceToLogin();
+
+    const li = parentSelector(voterEl, ".comment");
+    const scoreDiv = li.querySelector("div.score");
+    const formData = new FormData();
+    formData.append('reason', reason || '');
+    let showScore = true;
+    let score = parseInt(scoreDiv.innerHTML);
+    let action = "";
+
+    if (isNaN(score)) {
+      showScore = false;
+      score = 0;
+    }
+
+    if (li.classList.contains("upvoted") && point > 0) {
+      /* already upvoted, neutralize */
+      li.classList.remove("upvoted");
+      score--;
+      action = "unvote";
+    } else if (li.classList.contains("flagged") && point < 0) {
+      /* already flagged, neutralize */
+      li.classList.remove("flagged");
+      score++;
+      action = "unvote";
+    } else if (point > 0) {
+      if (li.classList.contains("flagged")) {
+        /* Give back the lost flagged point */
+        score++;
       }
+      li.classList.remove("flagged");
+      li.classList.add("upvoted");
+      score++;
+      action = "upvote";
+    } else if (point < 0) {
+      if (li.classList.contains("upvoted")) {
+        /* Removes the upvote point this user already gave the story*/
+        score--;
+      }
+      li.classList.remove("upvoted");
+      li.classList.add("flagged");
+      if (li.parentElement.querySelector('.comment_folder_button')) {
```

Comment:
Under what conditions would the folder be missing? When I look at `app/views/comments/_comment.html.erb` there's no conditional in there, every comment gets one.

If it's always there, remove this guard. If there's a place it might be missing, edit this + next line to extract the element to a variable rather than select it twice.